internal/target/repository_host_set.go

```diff
@@ -0,0 +1,349 @@
+package target
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+)
+
+// AddTargetHostSets provides the ability to add host sets (hostSetIds) to a
+// target (targetId).  The target's current db version must match the
+// targetVersion or an error will be returned.   The target and a list of
+// current host set ids will be returned on success. Zero is not a valid value
+// for the WithVersion option and will return an error.
+func (r *Repository) AddTargetHostSets(ctx context.Context, targetId string, targetVersion uint32, hostSetIds []string, _ ...Option) (Target, []*TargetSet, error) {
```

Comment:
As discussed for the TargetLibrary SAD operations, please also return []*CredentialLibrary here so the API can return the full Target resource.